retrofit/src/main/java/retrofit2/ServiceMethod.java

```diff
@@ -535,33 +536,47 @@ private Headers parseHeaders(String[] headers) {
           throw parameterError(p, "@Part parameters can only be used with multipart encoding.");
         }
         Part part = (Part) annotation;
-        Headers headers = Headers.of(
-            "Content-Disposition", "form-data; name=\"" + part.value() + "\"",
-            "Content-Transfer-Encoding", part.encoding());
+        gotPart = true;
 
+        String partName = part.value();
         Class<?> rawParameterType = Utils.getRawType(type);
-        gotPart = true;
-        if (Iterable.class.isAssignableFrom(rawParameterType)) {
-          if (!(type instanceof ParameterizedType)) {
-            throw parameterError(p, rawParameterType.getSimpleName()
-                + " must include generic type (e.g., "
-                + rawParameterType.getSimpleName()
-                + "<String>)");
+        if (partName.isEmpty()) {
+          if (!MultipartBody.Part.class.isAssignableFrom(rawParameterType)) {
+            throw parameterError(p,
+                "@Part annotation must supply a name or use MultipartBody.Part parameter type.");
           }
-          ParameterizedType parameterizedType = (ParameterizedType) type;
-          Type iterableType = Utils.getParameterUpperBound(0, parameterizedType);
-          Converter<?, RequestBody> converter = retrofit.requestBodyConverter(
-              iterableType, annotations, methodAnnotations);
-          return new ParameterHandler.Part<>(headers, converter).iterable();
-        } else if (rawParameterType.isArray()) {
-          Class<?> arrayComponentType = boxIfPrimitive(rawParameterType.getComponentType());
-          Converter<?, RequestBody> converter =
-              retrofit.requestBodyConverter(arrayComponentType, annotations, methodAnnotations);
-          return new ParameterHandler.Part<>(headers, converter).array();
+
+          return ParameterHandler.RawPart.INSTANCE;
         } else {
-          Converter<?, RequestBody> converter =
-              retrofit.requestBodyConverter(type, annotations, methodAnnotations);
-          return new ParameterHandler.Part<>(headers, converter);
+          Headers headers =
+              Headers.of("Content-Disposition", "form-data; name=\"" + partName + "\"",
```

Comment:
Probably want to eventually restrict this to not contain `"`, etc.
Or escape it somehow